packages/core/src/Cardano/types/Block.ts

```diff
@@ -74,18 +77,42 @@ export const SlotLeader = (value: string): SlotLeader => {
   }
 };
 
-export interface Block {
+/**
+ * Get Bech32 encoded VRF verification key from base64 encoded string
+ *
+ * @param value is a Base64 string
+ * @returns Bech32 encoded vrf_vk
+ */
+export const VrfVkBech32FromBase64 = (value: string) =>
+  VrfVkBech32(CSL.VRFVKey.from_bytes(Buffer.from(value, 'base64')).to_bech32('vrf_vk'));
+
+/** Minimal Block type that can be  */
+// REVIEW: made some of the properties `optional` because some may be undefined if block is Byron
+export interface BlockMinimal {
   header: PartialBlockHeader;
+  /** Byron blocks do not have fee */
+  fees?: Lovelace;
+  totalOutput: Lovelace;
+  txCount: number;
+  /** Byron blocks do not have size */
+  size?: BlockSize;
+  previousBlock?: BlockId;
+  vrf?: VrfVkBech32;
+}
+
+export interface Block extends BlockMinimal {
+  // In case of blocks produced by BFT nodes, the SlotLeader the issuerVk hash
+  // For blocks produced by stake pools, it is the Bech32 encoded value of issuerVk hash
+  slotLeader?: SlotLeader;
   date: Date;
   epoch: EpochNo;
   epochSlot: number;
-  slotLeader: SlotLeader;
-  size: BlockSize;
-  txCount: number;
-  totalOutput: Lovelace;
-  fees: Lovelace;
-  vrf: VrfVkBech32;
-  previousBlock?: BlockId;
   nextBlock?: BlockId;
   confirmations: number;
 }
+
+export interface BlockOgmios extends BlockMinimal {
```

Comment:
Would be best to not have ogmios-specific type here in `core`. I suggest hoisting `issuerVk` to `blockMinimal` and using that for translation util.

Reply:
I think I solved this problem. Please take a look at: [1fad239](https://github.com/input-output-hk/cardano-js-sdk/pull/450/commits/1fad2392e4bb48c94364c6fc6d9f7e618e6533ad)